.buildkite/release_pipeline.yaml

```diff
@@ -19,6 +19,14 @@ steps:
     command: .buildkite/scripts/build_component.sh hab
     agents:
       queue: habitat-release
+  
+  - label: ":linux: :see_no_evil: :habicat: core/hab/kernel2"
+    command: .buildkite/scripts/build_component.sh hab
+    concurrency_group: "habitat/release/x86_64-linux-kernel2"
```

Comment:
What's the intention behind the concurrency group for just the kernel2 builds? And also without a corresponding `concurrecy` key?

Reply:
That was in part a copy/paste from the bootstrap pipeline where we use it to enforce isolation.   I chatted with @scotthain about this after I initially published this branch and we decided we need to think on it a bit when we didn't have end-of-day pudding brain 😹   IIRC the question came down to: Do we need to ensure isolation between our two Linux builds for the release?

Reply:
I don't _think_ we do... the only kind of isolation I think we need is the sequencing, but that's taken care of using the various `wait` steps.

The only thing we're currently using concurrency on is the macOS builder, and then only because the builder is a single, long-lived machine, and we've had issues with our Travis-based release of having concurrent builds stepping on each other. It doesn't feel like there's a similar concern with the kernel 2 stuff, but I may be missing something; you guys have been more in that work than I have.

Reply:
☕️ Kicking in now.  My concern was that *if* a kernel-current and kernel2 build were scheduled on the same host, you would end up with failed or corrupted builds because `hab pkg build` is going to attempt to destroy any existing studio at the current path, which would be the same for the two Targets.    

We could probably work around this by naming the studio (`-r`) with the Target as part of the studio name, but it might just be safer to enforce them executing on separate hosts.  

I'm not sure if it's possible for builds to be scheduled on the same host, I'm not familiar enough with how it's set up, so this may be a non-concern.

Reply:
I think the path should be different, though, since they'd be different Buildkite jobs, and I think that's part of the path.

As for being scheduled on the same host, I think that depends on how the agent is configured, which is a question for @scotthain, methinks.

Reply:
After more thoughts, we shouldn't need the concurrency group in this case. (And if there's something we've manage to forget/nuke from our brains, we can add it back)

Reply:
Also as a side note, if we don't cut the concurrency down to 1 (as we do in some other pipelines as a forced limiting factor) there's no real point in this case :)